dascore/core/patch.py

```diff
@@ -30,16 +30,19 @@ def __init__(
         dims: Tuple[str] = None,
         attrs: Optional[Mapping] = None,
     ):
-        if isinstance(data, DataArray):
-            self._data_array = data
-            return
-        dims = dims if dims is not None else list(coords)
-        mixer = _AttrsCoordsMixer(attrs, coords, dims)
-        attrs, coords = mixer()
-        # get xarray coords from custom coords object
-        if isinstance(coords, Coords):
-            coords = coords._coords
-        self._data_array = DataArray(data=data, dims=dims, coords=coords, attrs=attrs)
+        if data is not None:  # be able to create empty patch instance to load file
+            if isinstance(data, DataArray):
+                self._data_array = data
+                return
+            dims = dims if dims is not None else list(coords)
+            mixer = _AttrsCoordsMixer(attrs, coords, dims)
+            attrs, coords = mixer()
+            # get xarray coords from custom coords object
+            if isinstance(coords, Coords):
+                coords = coords._coords
+            self._data_array = DataArray(
+                data=data, dims=dims, coords=coords, attrs=attrs
+            )
```

Comment:
Hey @jinwar, so I added a few tests in tests/core/patch.py. I decided to create a new test case (class) since there will likely be several tests around empty patches (a common edge case).

It looks like this:

```python
class TestEmptyPatch:
    """Tests for empty patch objects."""

    def test_empty_patch_str(self):
        """An empty patch should have both a str and repr of nonzero length."""
        patch = Patch()
        assert len(str(patch))
        assert len(repr(patch))

    def test_has_attrs(self):
        """An empty test patch should have attrs with null(ish) values."""
        patch = Patch()
        attrs = patch.attrs
        assert len(attrs)
        # Check that default values exist but are null (of appropriate dtype)
        names = ['time', 'distance']
        fields = ['d_{x}', '{x}_min', "{x}_max"]
        for name in names:
            for field in fields:
                value = attrs[field.format(x=name)]
                assert pd.isnull(value)
```

Reply:
I had to refactor the `Patch.__init__` method slightly to get the tests to pass.